src/proof/theory_proof.cpp

```diff
@@ -1264,4 +1263,387 @@ void TheoryProof::printRewriteProof(std::ostream& os, const Node &n1, const Node
+      {
+        for (unsigned j = 0; j < count; ++j)
+        {
+          if (pf.d_children[i + j]->d_children[0]->d_id
+              != theory::eq::MERGED_THROUGH_REFLEXIVITY)
+            orderedEqualities.insert(orderedEqualities.begin(),
+                                     pf.d_children[i + j]->d_children[0]);
+          if (pf.d_children[i + j]->d_children[1]->d_id
+              != theory::eq::MERGED_THROUGH_REFLEXIVITY)
+            orderedEqualities.insert(orderedEqualities.end(),
+                                     pf.d_children[i + j]->d_children[1]);
+        }
+      }
+      else
+      {
+        for (unsigned j = 0; j < count; ++j)
+        {
+          if (pf.d_children[i + count - 1 - j]->d_children[0]->d_id
+              != theory::eq::MERGED_THROUGH_REFLEXIVITY)
+            orderedEqualities.insert(
+                orderedEqualities.begin(),
+                pf.d_children[i + count - 1 - j]->d_children[0]);
+          if (pf.d_children[i + count - 1 - j]->d_children[1]->d_id
+              != theory::eq::MERGED_THROUGH_REFLEXIVITY)
+            orderedEqualities.insert(
+                orderedEqualities.end(),
+                pf.d_children[i + count - 1 - j]->d_children[1]);
+        }
+      }
+
+      // Copy the result into the main transitivity proof.
+      subTrans->d_children.insert(subTrans->d_children.end(),
+                                  orderedEqualities.begin(),
+                                  orderedEqualities.end());
+
+      // Increase i to skip over the children that have been processed.
+      i += count;
+      if (targetAppearsAfter)
+      {
+        ++i;
+      }
+    }
+
+    // Else, just copy the child proof as is
+    else
+    {
+      subTrans->d_children.push_back(pf.d_children[i]);
+      ++i;
+    }
+  }
+
+  bool disequalityFound = (*neg >= 0);
+  if (!disequalityFound)
+  {
+    Debug("pf::" + theoryName)
+        << "A disequality was NOT found. UNSAT due to merged constants"
+        << std::endl;
+    Debug("pf::" + theoryName) << "Proof for: " << pf.d_node << std::endl;
+    Assert(pf.d_node.getKind() == kind::EQUAL);
+    Assert(pf.d_node.getNumChildren() == 2);
+    Assert(pf.d_node[0].isConst() && pf.d_node[1].isConst());
+  }
+}
+
+void TheoryProof::identicalEqualitiesPrinterHelper(theory::TheoryId theoryId,
+                                            bool evenLengthSequence,
+                                            bool sequenceOver,
+                                            int i,
+                                            const theory::eq::EqProof& pf,
+                                            const ProofLetMap& map,
+                                            const Node& n2,
+                                            const std::string ss1String,
+                                            std::stringstream* ss,
+                                            Node& n1,
+                                            Node& nodeAfterEqualitySequence)
```

Comment:
Instead of passing in `n1` and `nodeAfterEqualitySequence` by reference, can we return a pair of nodes instead? We don't really pass around nodes by reference anywhere else (as far as I am aware of).